spring-beans/src/main/java/org/springframework/beans/propertyeditors/ZoneIdEditor.java

```diff
@@ -38,7 +39,13 @@ public void setAsText(String text) throws IllegalArgumentException {
 		if (StringUtils.hasText(text)) {
 			text = text.trim();
 		}
-		setValue(ZoneId.of(text));
+		try {
+
+			setValue(ZoneId.of(text));
+		}
+		catch (Exception ex) {
+			throw new IllegalArgumentException("Failed to convert ZoneId for " + text, ex);
```

Comment:
Other editors that throw exceptions wrap the `text` inside `'` chars, so `"Failed to convert ZoneId for '" + text + "'"` as the message. That way it is easy to identify trailing and/or leading spaces.